support/BRFileService.c

```diff
@@ -30,10 +30,75 @@
 #include <dirent.h>
 #include <sys/stat.h>
 #include <errno.h>
+#include <sqlite3.h>
+typedef int sqlite3_status_code;
 
 #define FILE_SERVICE_INITIAL_TYPE_COUNT    (5)
 #define FILE_SERVICE_INITIAL_HANDLER_COUNT    (2)
 
+#define FILE_SERVICE_SDB_FILENAME      "entities.db"
+
+#define FILE_SERVICE_SDB_ENTITY_TABLE     \
+"CREATE TABLE IF NOT EXISTS Entity(     \n\
+  Currency  CHAR(64)    NOT NULL,       \n\
+  Network   CHAR(64)    NOT NULL,       \n\
+  Type      CHAR(64)    NOT NULL,       \n\
+  Hash      TEXT        PRIMARY KEY,    \n\
+  Data      TEXT        NOT NULL);"
```

Comment:
For Data, why not use BLOB? For the primary key, is using TEXT going to be a performance issue at some point (unlikely, but just asking)?

Reply:
BLOB to avoid the hex encoding?  I do like seeing the hex-encoding in 'DB Query Tools' but admittedly have not seen how those tools (the one I use) displays BLOBs.

I know there is often concern about a non-integer PRIMARY KEY; but the Hash is the primary key; thus not much value in defining an unrelated integer value.  Is there a way to think of an integer as related?  Note: the actual primary key should be (Currency, Network, Type, Hash); can that be declared as such?

Reply:
You can have a primary key composed of multiple column values, IIRC.

Reply:
You can run a query to display BLOBs as hex. I understand the desire to see as hex-encoding, but it does have a (small) performance/size implication.